.azure-pipelines/ultimate-pipeline.yml

```
@@ -3154,6 +3154,29 @@ stages:
             AZURE_STORAGE_ACCOUNT: $(AZURE_STORAGE_ACCOUNT_NAME)
             AZURE_STORAGE_SAS_TOKEN: $(AZURE_STORAGE_SHARED_ACCESS_TOKEN)
 
+- stage: upload_container_images
+  condition: and(succeeded(), eq(variables['isBenchmarksOnlyBuild'], 'False'), eq(variables.isMainRepository, true), eq(variables.isMainOrReleaseBranch, true))
```

This might need looking at... I think `isBenchmarksOnlyBuild` is now _only_ false on the `benchmark/*` branches, it's _never_ false on `master`. That means this will run on _every_ build to master (i.e. it will run every night, _in addition_ to merges, and 10 times on Saturday).

Maybe the best option here is to explicitly only allow `IndividualCI` (i.e. merge commits) as the build reason...

```suggestion
  condition: and(succeeded(), eq(variables['Build.Reason'], 'IndividualCI'), eq(variables.isMainRepository, true), eq(variables.isMainOrReleaseBranch, true))
```

It's possible we have other stages where this is what we should be doing too... 😬

---

Ahhh I didn't see this until I merged, sorry. I created a small PR for this: https://github.com/DataDog/dd-trace-dotnet/pull/3783